src/main/java/org/ibp/api/java/impl/middleware/dataset/validator/InstanceValidator.java

```diff
@@ -1,36 +1,104 @@
 
-        if (!this.studyDataManager.existInstances(instanceIds)) {
+		if (!this.studyDataManager.existInstances(instanceIds)) {
 			this.errors.reject("dataset.non.existent.instances", "");
-            throw new ApiRequestValidationException(this.errors.getAllErrors());
-        }
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
 
 		if (datasetId != null && !this.studyDataManager.areAllInstancesExistInDataset(datasetId, instanceIds)) {
 			this.errors.reject("dataset.invalid.instances", "");
 			throw new ApiRequestValidationException(this.errors.getAllErrors());
 		}
 	}
 
+	public void validateForDesignGeneration(final Integer studyId, final Set<Integer> instanceNumbers) {
+		this.errors = new MapBindingResult(new HashMap<String, String>(), Integer.class.getName());
+
+		if (Collections.isEmpty(instanceNumbers)) {
+			this.errors.reject("study.instances.required");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+		final Map<String, Integer> instanceGeolocationIdsMap = this.studyDataManager.getInstanceGeolocationIdsMap(studyId);
+		final List<String> instanceNumberStringList = instanceNumbers.stream().map(s -> s.toString()).collect(Collectors.toList());
+		if (instanceGeolocationIdsMap == null || instanceGeolocationIdsMap.isEmpty() || !instanceGeolocationIdsMap.keySet()
+			.containsAll(instanceNumberStringList)) {
+			this.errors.reject("dataset.non.existent.instances");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+
+		final List<StudyInstance> studyInstances = this.studyInstanceService.getStudyInstances(studyId);
+		final List<Integer> restrictedInstances = new ArrayList<>();
+		final List<Integer> instancesWithDesign = new ArrayList<>();
+		for (final StudyInstance instance : studyInstances) {
+			if (BooleanUtils.isFalse(instance.getCanBeDeleted())) {
+				restrictedInstances.add(instance.getInstanceNumber());
+			}
+			if (BooleanUtils.isTrue(instance.isHasExperimentalDesign())) {
+				instancesWithDesign.add(instance.getInstanceNumber());
+			}
+		}
+
+		// Check that at least one instance is not restricted from design regeneration
+		if (restrictedInstances.containsAll(instanceNumbers)) {
+			this.errors.reject("all.selected.instances.cannot.be.regenerated");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+
+		// If at least one instance is for regeneration, check that study has no advanced or crosses list
+		if (!instanceNumbers.stream()
+			.distinct()
+			.filter(instancesWithDesign::contains)
+			.collect(Collectors.toSet()).isEmpty() && this.middlewareStudyService.hasAdvancedOrCrossesList(studyId)) {
+			this.errors.reject("study.has.advance.or.cross.list");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	public void checkStudyInstanceAlreadyExists(final Integer studyId, final int instanceNumber) {
```

Comment:
Same, we don't need this validation if we don't expose the instance number as parameter

Reply:
Done